src/store/reducers/UserProfile.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  UserProfileActions,
+  UserProfileActionTypes,
+} from "store/actions/UserProfile";
+
+export type UserProfileState = {
+  isUserProfileVisible: boolean;
```

Comment:
I would probably just simplify this to `isVisible` as well. We already know that it's in the context of the `UserProfileState`, so adding that again in the name here feels redundant.